### PR TITLE
Slashing state transaction creation and validation

### PR DESCRIPTION
**Pull request by dusan-maksimovic**
# Description

Introduced slashing state transaction to be added to the block when double signing validators are detected for previous block.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

This PR changes the consensus so it must be introduced through the hard fork. New hard fork is added to the code, it is called "doubleSignSlashing"

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

**Comment by Nemanja0x** [comment]
It looks good, just need to fix linters.

File: consensus/polybft/fsm.go [hunk 1]
```diff
@@ -467,11 +509,30 @@ func (f *fsm) VerifyStateTransactions(transactions []*types.Transaction) error {
...
+			}
+
+			if slashingTxExists {
+				return errSlashingTxSingleExpected
+			}
+
+			slashingTxExists = true
+
+			if err := f.verifySlashingTx(tx); err != nil {
+				return fmt.Errorf("error while verifying slashing transaction. error: %w", err)
```
Comment (Stefan-Ethernal) [inline consensus/polybft/fsm.go]: Nitpick (consider if we can better format error message)
```suggestion
				return fmt.Errorf("error while verifying slashing transaction: %w", err)
```

File: consensus/polybft/fsm.go [hunk 2]
```diff
@@ -235,6 +250,32 @@ func (f *fsm) createBridgeCommitmentTx() (*types.Transaction, error) {
...
+		return nil
+	}
+
+	slashingTx, err := f.createSlashingTx()
+	if err != nil {
+		return fmt.Errorf("creation of slashing transaction failed: %w", err)
+	}
+
+	if err = f.blockBuilder.WriteTx(slashingTx); err != nil {
+		return fmt.Errorf("failed to apply slashing state transaction. Error: %w", err)
```
Comment (Stefan-Ethernal) [inline consensus/polybft/fsm.go]: Nitpick (to be consistent with other error messages):
```suggestion
		return fmt.Errorf("failed to apply slashing state transaction: %w", err)
```

File: consensus/polybft/fsm.go [hunk 3]
```diff
@@ -235,6 +250,32 @@ func (f *fsm) createBridgeCommitmentTx() (*types.Transaction, error) {
 	return createStateTransactionWithData(f.Height(), contracts.StateReceiverContract, inputData), nil
 }
 
+func (f *fsm) applySlashingTx() error {
```
Comment (Stefan-Ethernal) [inline consensus/polybft/fsm.go:254]: Consider adding comments on newly added functions.

**Review by Stefan-Ethernal — APPROVED** [review]

**Review by Nemanja0x — APPROVED** [review]

File: consensus/polybft/fsm.go (hunk 1 above)
Comment (dusan-maksimovic) [reply consensus/polybft/fsm.go]: Fixed [3a65efb](https://github.com/0xPolygon/polygon-edge/pull/1815/commits/3a65efb21cb39f9cc4f50c963d86c19959e114dd)

File: consensus/polybft/fsm.go (hunk 2 above)
Comment (dusan-maksimovic) [reply consensus/polybft/fsm.go]: Fixed [3a65efb](https://github.com/0xPolygon/polygon-edge/pull/1815/commits/3a65efb21cb39f9cc4f50c963d86c19959e114dd)

File: consensus/polybft/fsm.go (hunk 3 above)
Comment (dusan-maksimovic) [reply consensus/polybft/fsm.go:254]: Fixed [3a65efb](https://github.com/0xPolygon/polygon-edge/pull/1815/commits/3a65efb21cb39f9cc4f50c963d86c19959e114dd)

**Pull request merged**